src/transaction/simple.rs

```diff
@@ -0,0 +1,271 @@
+//!
+//! A generic transaction API. Intended to work with
+//! multiple connection types/variations.
+//!
+
+use crate::connection::simple::TypeConnectionContainer;
+use crate::{Execute, FbError, FromRow, IntoParams, Queryable, SimpleConnection, Transaction};
+#[cfg(feature = "linking")]
+use rsfbclient_native::DynLink;
+#[cfg(feature = "dynamic_loading")]
+use rsfbclient_native::DynLoad;
+#[cfg(any(feature = "linking", feature = "dynamic_loading"))]
+use rsfbclient_native::NativeFbClient;
+#[cfg(feature = "pure_rust")]
+use rsfbclient_rust::RustFbClient;
+use std::convert::From;
+
+/// A transaction API without client types
+pub struct SimpleTransaction<'c> {
+    inner: TypeTransactionContainer<'c>,
+}
+
+enum TypeTransactionContainer<'c> {
+    #[cfg(feature = "linking")]
+    NativeDynLink(Transaction<'c, NativeFbClient<DynLink>>),
+    #[cfg(feature = "dynamic_loading")]
+    NativeDynLoad(Transaction<'c, NativeFbClient<DynLoad>>),
+    #[cfg(feature = "pure_rust")]
+    PureRust(Transaction<'c, RustFbClient>),
+}
+
+#[cfg(feature = "linking")]
+impl<'c> From<Transaction<'c, NativeFbClient<DynLink>>> for SimpleTransaction<'c> {
+    fn from(tr: Transaction<'c, NativeFbClient<DynLink>>) -> Self {
+        let inner = TypeTransactionContainer::NativeDynLink(tr);
+        SimpleTransaction { inner }
+    }
+}
+
+#[cfg(feature = "linking")]
+impl<'c> From<SimpleTransaction<'c>> for Result<Transaction<'c, NativeFbClient<DynLink>>, FbError> {
```

Comment:
why not `TryFrom`?

Reply:
Because I forgot about `TryFrom` hehe. I will fix.